chainer/functions/loss/crf1d.py

```diff
@@ -0,0 +1,87 @@
+from chainer.functions.array import broadcast
+from chainer.functions.array import reshape
+from chainer.functions.array import select_item
+from chainer.functions.connection import embed_id
+from chainer.functions.math import logsumexp
+from chainer.functions.math import minmax
+from chainer.functions.math import sum as _sum
+
+
+def crf1d(cost, xs, ys):
+
+    """Calculates negative log-likelihood of linear-chain CRF.
+
+    It takes a transition cost matrix, a sequence of costs and a sequence of
+    labels. Let :math:`c_{st}` is transition cost from a label :math:`s` to
+    a label :math:`t`, :math:`x_{it}` is a cost value of a label :math:`t` at
+    position :math:`i`, and :math:`y_i` is an expected label value at position
+    :math:`i`. The negative log-likelihood of linear-chain CRF is defined as
```

Comment:
It takes a transition cost matrix, a sequence of costs, and a sequence of labels.
Let :math:`c_{st}` be a transition cost from a label :math:`s` to a label :math:`t`,
:math:`x_{it}` be a cost of a label :math:`t` at position :math:`i`, and
:math:`y_i` be an expected label at position :math:`i`.